source/common/common/callback_impl.h

```diff
@@ -61,8 +63,8 @@ template <typename... CallbackArgs> class CallbackManager {
                         [handle](const CallbackHolder& holder) -> bool {
                           return handle == &holder;
                         }) != callbacks_.end());
-    callbacks_.remove_if(
-        [handle](const CallbackHolder& holder) -> bool { return handle == &holder; });
+    auto it = static_cast<CallbackHolder*>(handle)->it_;
```

Comment:
make it a dynamic cast

Reply:
Thank you for pointing it out. I should make it dynamic cast because the two class has virtual function.